internal/store/migrations.go

```diff
@@ -984,6 +984,236 @@ var migrations = []migration{
 			return err
 		}
 
+		return nil
+	}},
+	{semver.MustParse("0.19.0"), semver.MustParse("0.20.0"), func(e execer) error {
+		// Changes:
+		// 1. Add APISecurityLock to cluster, installation, cluster installation
+		//    and group resources.
+		// 2. Remove deprecated cluster fields.
+
+		_, err := e.Exec(`ALTER TABLE Cluster RENAME TO ClusterTemp;`)
+		if err != nil {
+			return err
+		}
+
+		_, err = e.Exec(`
+			CREATE TABLE Cluster (
+				ID TEXT PRIMARY KEY,
+				Provider TEXT NOT NULL,
+				Provisioner TEXT NOT NULL,
+				ProviderMetadataRaw BYTEA NULL,
+				ProvisionerMetadataRaw BYTEA NULL,
+				UtilityMetadataRaw BYTEA NULL,
+				State TEXT NOT NULL,
+				AllowInstallations BOOLEAN NOT NULL,
+				CreateAt BIGINT NOT NULL,
+				DeleteAt BIGINT NOT NULL,
+				APISecurityLock BOOLEAN NOT NULL,
+				LockAcquiredBy TEXT NULL,
+				LockAcquiredAt BIGINT NOT NULL
+			);
+		`)
+		if err != nil {
+			return err
+		}
+
+		_, err = e.Exec(`
+			INSERT INTO Cluster
+			SELECT
+				ID,
+				Provider,
+				Provisioner,
+				ProviderMetadataRaw,
+				ProvisionerMetadataRaw,
+				UtilityMetadataRaw,
+				State,
+				AllowInstallations,
+				CreateAt,
+				DeleteAt,
+				'false',
+				LockAcquiredBy,
+				LockAcquiredAt
+			FROM
+				ClusterTemp;
+		`)
+		if err != nil {
+			return err
+		}
+
+		_, err = e.Exec(`DROP TABLE ClusterTemp;`)
```

Comment:
I thought that both of our database engines supported using ALTER TABLE to add Columns. If this is the case, I would like to see us use that instead of copying data to a temporary table to add columns. These copies are going to really add up when there's a lot of data in this database and since the migrations represent the schema and are therefore a permanent fixture of the process, if we keep using full table copies to add columns the migration process will eventually get very lengthy.

In other words, couldn't lines 989 to 1044 be replaced with
```go
_, err := e.Exec(`ALTER TABLE Cluster ADD COLUMN ApiSecurityLock BOOLEAN NOT NULL DEFAULT 'false';`)
if err != nil {
    return err
}
```
?

Reply:
(also, did we drop support for sqlite? I seem to remember it being discussed awhile back but I've been testing with a local Postgres instance for so long that now I can't recall if we actually dropped the support for SQLite or not. If we did drop support for it, I think Postgres lets you use `ALTER TABLE` like I've described)

Reply:
Nope! We didn't drop SQLite support (yet). It is still somewhat useful so I am dragging it along still, but maybe this should be where we stop supporting it.

Regardless, good point on simplifying the migration. The cluster table still is probably better suited doing the original flow as it also drops two columns in the process and splitting up that logic for mysql and PG seems to be a bit messy, but the other tables are now updated as you recommended.